airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/source.py

```diff
@@ -420,11 +416,35 @@ def check_for_sampled_result(self, data: Mapping) -> None:
 class GoogleAnalyticsV4IncrementalObjectsBase(GoogleAnalyticsV4Stream):
     cursor_field = "ga_date"
 
-    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
```

Comment:
stream should be able to return state after the sync. it is logged in output and then will be re-used in next syncs

Reply:
could you please describe what was working wrong in the current implementation ?

Reply:
`state` getter and setter are new style implementation of incremental streams instead of deprecated `get_updated_state`.

The main defect was `stream_slices` returning {`startDate`:`<today>`, `endDate`:`<today>`} slice for future state, and `read_records` was emitting respective records, although it should not emit any.

Reply:
rolled back to `get_updated_state`